### PR TITLE
ci: drop pouchdb-server from test matrix

### PR DESCRIPTION
**Pull request by alxndrsn**
* `pouchdb-server` hasn't had an update in 3 years; hasn't had a release in 5 years (https://github.com/pouchdb/pouchdb-server/commits/master/)
* seems to be the blocker for updating the base node version for builds from node v14 (e.g. https://github.com/pouchdb/pouchdb/pull/8664)
* moving the `pouchdb-server` build from travis -> github actions and making it pass seems to be non-trivial (https://github.com/alxndrsn/pouchdb-server/actions/runs/8410590833)

### CODE REVIEW

**Comment by alxndrsn** [comment]
Should `browser-adapter` tests also be dropped?

**Comment by garethbowen** [comment]
> Should browser-adapter tests also be dropped?

Why? Do they hit pouch-server too?

**Comment by alxndrsn** [comment]
> > Should browser-adapter tests also be dropped?
> 
> Why? Do they hit pouch-server too?

They use `pouchdb-express-router`.  I think I was confused - there are 3 pouch-based servers currently referenced from this repo:

1. `pouchdb-server`: https://github.com/pouchdb/pouchdb-server
2. `pouchdb-express-router`: https://github.com/pouchdb/pouchdb-express-router
3. `express-pouchdb-minimum`: https://github.com/pouchdb/pouchdb-server/tree/master/packages/node_modules/express-pouchdb

1 & 3 are based on `pouchdb-server`; #2 seems to be separate.

File: TESTING.md [hunk 1]
```diff
@@ -151,10 +151,6 @@ following values:
 - `couchdb-master`: use this value if you already have CouchDB running; it
   causes `COUCH_HOST` to be set to the correct value.
 
-#### `SKIP_MIGRATION` (default: `0`)
```
Comment (janl) [inline TESTING.md:154]: why is this removed?

Comment (alxndrsn) [reply TESTING.md:154]: This PR is in draft; reviewing it may be confusing at this time.

**Comment by alxndrsn** [comment]
here's a quick summary of the most recent failures seen on `master` as per 2024-04-09:

```json
{
  "couchdb-nodejs": {
    "_total_errors": 2,
    "3.1, 16, npm test": 2
  },
  "pouchdb-server": {
    "_total_errors": 17,
    "node, npm test": 12,
    "webkit, npm test": 1,
    "chromium, npm test": 2,
    "firefox, npm test": 2
  },
  "browser-adapter": {
    "_total_errors": 69,
    "chromium, idb, npm test": 1,
    "webkit, idb, npm test": 7,
    "webkit, indexeddb, npm test": 5,
    "webkit, memory, npm test": 16,
    "firefox, idb, npm test": 6,
    "firefox, memory, npm test": 17,
    "firefox, indexeddb, npm test": 16,
    "chromium, indexeddb, npm test": 1
  },
  "nodejs": {
    "_total_errors": 34,
    "16, CLIENT=firefox npm run test-webpack": 18,
    "14, CLIENT=firefox npm run test-webpack": 16
  },
  "nodejs-adapter": {
    "_total_errors": 5,
    "14, memory, npm test": 2,
    "16, leveldb, npm test": 1,
    "16, memory, npm test": 2
  },
  "couchdb-browser": {
    "_total_errors": 1,
    "2.3, webkit, npm test": 1
  }
}
```

`browser-adapter` tests run against `pouchdb-express-router`, not `pouchdb-server`.

Dropping both of these might be an option, but it's unclear if test flake is actually:

1. in pouch core code
2. in pouch adapter code
3. in pouch test code
4. in server implementations
